sunpy/visualization/mapcubeanimator.py

```diff
@@ -33,15 +33,26 @@ class MapCubeAnimator(imageanimator.BaseFuncAnimator):
     colorbar: bool
         Plot colorbar
 
+    plot_function: function
+        A function to call when each map is plotted, the function must have
+        the signature `(fig, axes, map)` where fig and axes are the figure and
+        axes objects of the plot and map is the current frames Map object.
+        Any objects returned from this function will have their `remove()` method
+        called at the start of the next frame to clear them from the plot.
+
     Notes
     -----
     Extra keywords are passed to `mapcube[0].plot()` i.e. the `plot()` routine of
-    the first map in the cube.
+    the maps in the cube.
     """
     def __init__(self, mapcube, annotate=True, **kwargs):
 
         self.mapcube = mapcube
         self.annotate = annotate
+        self.user_plot_function = kwargs.pop('plot_function',
+                                             lambda fig, ax, map: [])
```

Comment:
'map' is a built-in function in Python.  Should probably use another variable name instead of map.